File: sdk/kotlin/src/main/kotlin/io/cloudmock/BLEMesh.kt
// BLEMesh — Bluetooth Low Energy mesh topology discovery for Android
//
// Discovers nearby devices running the CloudMock SDK via BLE advertising
// and builds a local mesh topology map. This enables:
//   - Device-to-device latency measurement
//   - Proximity-based service discovery
//   - Offline mesh networking between Android devices
//   - Real-time topology visualization in the CloudMock dashboard
//
// The mesh uses BLE peripheral/central roles simultaneously:
//   - Advertises a CloudMock service UUID with device metadata
//   - Scans for nearby CloudMock devices and measures RSSI
//   - Reports discovered topology to the CloudMock endpoint
//
// Usage:
//   // Enabled via CloudMock.initialize() when enableBLE = true
//   // Or manually:
//   BLEMesh.start(context)
//   val peers = BLEMesh.discoveredPeers()
package io.cloudmock

/**
 * BLEPeer represents a discovered device in the BLE mesh.
 */
data class BLEPeer(
    val deviceID: String,
    val appName: String,
    val rssi: Int,
    val lastSeenMs: Long
)

/**
 * BLEMesh provides Bluetooth mesh topology discovery for nearby devices.
 */
object BLEMesh {

    /** Start BLE mesh discovery. Requires Bluetooth and location permissions. */
    fun start(/* context: Context */) {
        // TODO: Initialize BluetoothLeAdvertiser and BluetoothLeScanner.
        // Advertise CloudMock service UUID. Scan for peers.
    }

    /** Stop BLE mesh discovery. */
    fun stop() {
        // TODO: Stop BLE advertising and scanning, clean up peers.
    }

    /** Returns the list of currently discovered peer devices. */
    fun discoveredPeers(): List<BLEPeer> {
        // TODO: Return current peer list from scan results.
        return emptyList()
    }

    /** Report the current mesh topology to the CloudMock endpoint. */
    fun reportTopology() {
        // TODO: Serialize peer list and RSSI data, POST to endpoint.
    }
}

File: sdk/swift/Sources/CloudMock/BLEMesh.swift
// BLEMesh — Bluetooth Low Energy mesh topology discovery
//
// Discovers nearby devices running the CloudMock SDK via BLE advertising
// and builds a local mesh topology map. This enables:
//   - Device-to-device latency measurement
//   - Proximity-based service discovery
//   - Offline mesh networking between mobile devices
//   - Real-time topology visualization in the CloudMock dashboard
//
// The mesh uses BLE peripheral/central roles simultaneously:
//   - Advertises a CloudMock service UUID with device metadata
//   - Scans for nearby CloudMock devices and measures RSSI
//   - Reports discovered topology to the CloudMock endpoint
//
// Usage:
//   // Enabled via CloudMock.initialize(config:) when enableBLE = true
//   // Or manually:
//   BLEMesh.start()
//   let peers = BLEMesh.discoveredPeers()

import Foundation

/// BLEMesh provides Bluetooth mesh topology discovery for nearby devices.
public class BLEMesh {

    /// Start BLE mesh discovery. Begins advertising and scanning.
    /// Requires Bluetooth permission from the user.
    public static func start() {
        // TODO: Initialize CBCentralManager and CBPeripheralManager.
        // Advertise CloudMock service UUID. Scan for peers.
    }

    /// Stop BLE mesh discovery. Stops advertising and scanning.
    public static func stop() {
        // TODO: Stop BLE managers, clean up discovered peers.
    }

    /// Returns the list of currently discovered peer devices.
    public static func discoveredPeers() -> [BLEPeer] {
        // TODO: Return current peer list from scan results.
        return []
    }

    /// Report the current mesh topology to the CloudMock endpoint.
    public static func reportTopology() {
        // TODO: Serialize peer list and RSSI data, POST to endpoint.
    }
}

/// BLEPeer represents a discovered device in the BLE mesh.
public struct BLEPeer {
    public let deviceID: String
    public let appName: String
    public let rssi: Int
    public let lastSeen: Date
}
